packages/compass-workspaces/src/components/workspace-tab-state-provider.tsx

```diff
@@ -12,6 +12,10 @@ import { createStore } from 'redux';
 
 type TabState = Record<string, Record<string, unknown>>;
 
+export type TableDataObject = {
+  columnWidths: Record<string, number>;
+};
```

Comment:
This breaks the logical separation we're keeping between packages. Workspaces shouldn't know about exact details of the CRUD state. If you need to persist tab state, the `useTabState` hook is already exposed from the workspaces package. Please move this very specific state from the workspaces plugin to crud as this is the only place where it is relevant. Implementing this functionality shouldn't require you to make any changes to compass-workspaces